src/main/java/at/aau/se2/handler/game/subhandlers/RequestUsernamesForSwitchHandler.java

```diff
@@ -15,16 +15,17 @@
 
 public class RequestUsernamesForSwitchHandler extends RequestUsernamesHandler{
 
-    ArrayList<String> usernames = new ArrayList<String>();
+    public ArrayList<String> usernames = new ArrayList<String>();
     @Override
     public void handleMessage(WebSocketSession session, JsonNode msg, Lobby lobby) {
         try{
-            List<Player> players = GameHandler.getPlayersofGame();
-            //hier dafür sorgen dass eigener username entfernt wird
+            List<Player> players = List.copyOf(GameHandler.getPlayersofGame());
+
             for(Player a:players){
-                if(!a.getSession().equals(session)){
-                    usernames.add(a.getUsername());
+                if(a.getSession().equals(session)){
+
                 }
+                else{usernames.add(a.getUsername());}
```

Comment:
Wenn if-Zweig leer und nur else-Zweig => Bedingung ändern

```suggestion
                if(!a.getSession().equals(session)){
                    usernames.add(a.getUsername());
                }
```

Reply:
ok